react/react1/week1/todolist/src/components/Table.js

```diff
@@ -0,0 +1,75 @@
+import { useState } from "react";
+import { useEffect } from "react";
+import ItemRow from "./ItemRow";
+import Header from "./Header";
+const todos = [
+  {
+    id: 1,
+    description: "Get out of bed",
+    done: false,
+  },
+  {
+    id: 2,
+    description: "Brush teeth",
+    done: false,
+  },
+  {
+    id: 3,
+    description: "Eat breakfast",
+    done: false,
+  },
+];
+
+function Table() {
+  const [todoList, setTodoList] = useState(todos);
+  const [count, setCount] = useState(0);
+  function deleteCounter(deleteId) {
+    setTodoList(todoList.filter(({ id }) => id !== deleteId));
+  }
+  function changeStatus(doneId) {
+    const newState = todoList.map((obj) => {
+      if (obj.id === doneId) {
+        return { ...obj, done: !obj.done };
+      }
+      return obj;
+    });
+
+    setTodoList(newState);
+  }
+
+  function addTodo() {
+    setTodoList(
+      todoList.concat([
```

Comment:
1 - you could also destructure the array and simply add a new item
2 - this id generator breaks your code if you remove an item from the middle of the array, so you need to create an exclusive id, you can use the inbuilt Date.now(), for example:
```
setState(...state, {
id: Date.now(),
todo: "review Andrii's PR"
})
```